### PR TITLE
Protect against empty packet lists

### PR DESCRIPTION
**Pull request by Atlante45**
Test plan:
- Standard smoke test

https://highfidelity.fogbugz.com/f/cases/2137/crash-in-udt-Packet-writeMessageNumber

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2cd9cf6ff1e0cfc37829d2acb07b872af66eae0a available at: https://gist.github.com/f5568f3b67a9e4483a65e83868ba1e52<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9056-2cd9cf6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9056-2cd9cf6.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10
QA Pass
General smoke test passes.  Also connected second machine to primary running this stack and all good.

File: libraries/networking/src/udt/Socket.cpp [hunk 1]
```diff
@@ -165,6 +165,12 @@ qint64 Socket::writePacketList(std::unique_ptr<PacketList> packetList, const Hif
         // hand this packetList off to writeReliablePacketList
         // because Qt can't invoke with the unique_ptr we have to release it here and re-construct in writeReliablePacketList
 
+        if (packetList->getNumPackets() == 0) {
+            qCWarning(networking) << "Trying to sent packet list with 0 packets, bailing.";
```
Comment (huffman) [inline libraries/networking/src/udt/Socket.cpp]: `sent` -> `send` but not a big deal

Comment (Atlante45) [inline libraries/networking/src/udt/Socket.cpp]: Will fix

**Comment by hifi-gustavo** [comment]
Build console output for commit: a38846aeed9467ccb3283930840fa48a1e285c14 available at: https://gist.github.com/f5568f3b67a9e4483a65e83868ba1e52<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9056-a38846a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9056-a38846a.dmg

**Pull request merged**